### DISCUSSION TITLE
Workspace links

### DISCUSSION THREAD

**Discussion by Eomm** [npm/rfcs#485] [Question] [answered]
Hi,
reading [the arborist concepts](https://github.com/npm/arborist/blob/main/notes/workspace.md#registry-vs-workspace-relationship) I was wondering how could I declare a workspace dependency?

Let's make an example:

```
workspace-root
+-- packages
    +-- a
    +-- b 
    +-- c 
```

all the projects are linked to each other by default.
I want to declare that the workspace `a` and the workspace `b` depends on `c`.

Running:

```
npm install c -w a -w b
```

lead to installing an npm's module fetch from the registry.

yarn has the `workspace:c` syntax to archive the desired behaviour.

Is it possible to do the same within npm workspaces?

**Comment by darcyclarke** [comment, marked as answer]
There's a couple ways to do this. The first is that you can specify a relative path (if you're working locally) vs. using the name of the workspace (since npm will try to fetch from the registry - as you noticed) **or** you can you use `npm link` (ref. https://docs.npmjs.com/cli/v7/commands/npm-link#workspace-usage & https://docs.npmjs.com/cli/v7/using-npm/workspaces). In both cases you'll want to add the dep but not install it. Since we don't have a great way of doing that today, you can use `npm pkg set`.

#### Example 1 - relative path

```bash
$ npm pkg set dependencies[c]="file:../c" -w a -w b
$ npm install
```

#### Example 2 - link

```bash
$ npm pkg set dependencies[c]="*" -w a -w b
$ npm link -w c
$ npm link c -w a -w b
$ npm install
```

I mocked these out quickly so let me know if you run into any issues with either strategy

**Reply by Eomm** [reply]
Clear, I thought the npm workspaces aim to avoid the `npm link` command tbh.


I think the second script did the job, with one small fix:

```
npm pkg set dependencies.c="*" -w a -w b
```

Thanks
